packages/widgets/src/HeaderBar/Profile.js

```diff
@@ -49,34 +42,35 @@ export default class Profile extends React.Component {
 
         return (
             <TextIcon
-                name={me.name}
+                name={name}
                 onClick={this.onToggle}
                 dataTestId="headerbar-profile-icon-text"
             />
         )
     }
 
     render() {
-        const { user, baseUrl, helpUrl } = this.props
+        const { name, email, avatar, baseUrl, helpUrl } = this.props
 
         return (
             <div
                 ref={c => (this.elContainer = c)}
                 data-test="headerbar-profile"
+                className="headerbar-profile"
             >
-                {this.userIcon(user, baseUrl)}
+                {this.renderUserIcon({ avatar, name, baseUrl })}
 
                 {this.state.show ? (
                     <ProfileMenu
-                        avatar={avatarPath(user.avatar, baseUrl)}
-                        name={user.name}
-                        email={user.email}
+                        avatarUrl={avatar ? avatarUrl(avatar, baseUrl) : null}
```

Comment:
I think I'd prefer to keep the complexity in the function instead of pulling it out.